### PR TITLE
Feat/met 5637 Implement new dataset info endpoint

### PR DESCRIPTION
**Pull request by JoanaCMS**

### CODE REVIEW

File: src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java [hunk 1]
```diff
@@ -89,6 +93,14 @@ public boolean isXsltPresent(String datasetId) {
     return datasetRepository.isXsltPresent(Integer.parseInt(datasetId)) != 0;
   }
 
+  @Override
+  public DatasetInfoDto getDatasetInfo(String datasetId) {
+    DatasetEntity datasetEntity = datasetRepository.findById(Integer.valueOf(datasetId)).orElseThrow(() -> new InvalidDatasetException(datasetId));
+    return new DatasetInfoDto(datasetId, datasetEntity.getDatasetName(), datasetEntity.getCreatedDate(), datasetEntity.getLanguage(),
+            datasetEntity.getCountry(), getHarvestingParameterDto(datasetId), isXsltPresent(datasetId));
+
```
Comment (jeortizquan) [inline src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java:101]: suggestion to remove this whitespace enter.

File: src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerIT.java [hunk 2]
```diff
@@ -194,18 +194,44 @@ public void retrieveDataset_expectStatus_ok() throws IOException {
...
 
     ResponseEntity<String> getDatasetResponse =
-        testRestTemplate.getForEntity(getBaseUrl() + "/dataset/{id}", String.class, expectedDatasetId);
+        testRestTemplate.getForEntity(getBaseUrl() + "/dataset/{id}/progress", String.class, expectedDatasetId);
+
+    assertEquals(HttpStatus.OK, getDatasetResponse.getStatusCode());
+    assertNotNull(getDatasetResponse.getBody());
+    JSONAssert.assertEquals(StringUtils.deleteWhitespace(datasetResponseBodyContent),
+        StringUtils.deleteWhitespace(getDatasetResponse.getBody()), true);
+
```
Comment (jeortizquan) [inline src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerIT.java:207]: clean up white space.

File: src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerIT.java [hunk 3]
```diff
@@ -194,18 +194,44 @@ public void retrieveDataset_expectStatus_ok() throws IOException {
...
+            testRestTemplate.getForEntity(getBaseUrl() + "/dataset/{id}/info", String.class, expectedDatasetId);
 
     assertEquals(HttpStatus.OK, getDatasetResponse.getStatusCode());
     assertNotNull(getDatasetResponse.getBody());
     assertTrue(getDatasetResponse.getBody().contains("\"creation-date\""));
-    JSONAssert.assertEquals(
-        expectedDatasetInfoJson(StringUtils.deleteWhitespace(datasetResponseBodyContent), String.valueOf(expectedDatasetId)),
-        StringUtils.deleteWhitespace(removeCreationDate(getDatasetResponse.getBody())), true);
+    JSONAssert.assertEquals(StringUtils.deleteWhitespace(datasetResponseBodyContent),
+            StringUtils.deleteWhitespace(removeCreationDate(getDatasetResponse.getBody())), true);
```
Comment (jeortizquan) [inline src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerIT.java:235]: clean up white space.

File: src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerTest.java [hunk 4]
```diff
@@ -483,64 +482,134 @@ void processDatasetFromOAI_differentXsltFileType_expectFail() throws Exception {
...
         .andExpect(jsonPath("$.progress-by-step[1].errors[0].message",
-            is(message1)))
-        .andExpect(jsonPath("$.dataset-info.dataset-id", is("12345")))
-        .andExpect(jsonPath("$.dataset-info.dataset-name", is("Test")))
-        .andExpect(jsonPath("$.dataset-info.creation-date", is("-999999999-01-01T00:00:00")))
-        .andExpect(jsonPath("$.dataset-info.language", is("Dutch")))
-        .andExpect(jsonPath("$.dataset-info.country", is("Netherlands")))
-        .andExpect(jsonPath("$.dataset-info.record-limit-exceeded", is(false)))
-        .andExpect(jsonPath("$.dataset-info.transformed-to-edm-external", is(false)));
+            is(message1)));
```
Comment (jeortizquan) [inline src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerTest.java:505]: my suggestion is to add this missing check as well.
```
 .andExpect(jsonPath("$.progress-by-step[1].errors[1].message",
           is(message2)));
```

**Review by jeortizquan — COMMENTED** [review]
Is working as intended just small clean ups are suggested and a test verification

File: src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetReportServiceImpl.java [hunk 5]
```diff
@@ -111,12 +101,15 @@ public ProgressInfoDto getReport(String datasetId) {
         // get qty of records completely processed
         long completedRecords = getCompletedRecords(stepStatistics);
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetReportServiceImpl.java:102]: Done

File: src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java [hunk 6]
```diff
@@ -4,11 +4,14 @@
 import static java.util.Objects.requireNonNull;
 import static java.util.stream.Collectors.toList;
 
+import eu.europeana.metis.sandbox.common.exception.InvalidDatasetException;
 import eu.europeana.metis.sandbox.common.exception.ServiceException;
 import eu.europeana.metis.sandbox.common.exception.XsltProcessingException;
 import eu.europeana.metis.sandbox.common.locale.Country;
 import eu.europeana.metis.sandbox.common.locale.Language;
+import eu.europeana.metis.sandbox.dto.*;
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java:12]: Fixed

File: src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerTest.java [hunk 7]
```diff
@@ -38,8 +38,7 @@
 import eu.europeana.metis.sandbox.common.locale.Language;
 import eu.europeana.metis.sandbox.controller.ratelimit.RateLimitInterceptor;
 import eu.europeana.metis.sandbox.domain.DatasetMetadata;
-import eu.europeana.metis.sandbox.dto.DatasetInfoDto;
-import eu.europeana.metis.sandbox.dto.RecordTiersInfoDto;
+import eu.europeana.metis.sandbox.dto.*;
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerTest.java:41]: Fixed

File: src/test/java/eu/europeana/metis/sandbox/service/problempatterns/PatternAnalysisServiceImplIT.java [hunk 8]
```diff
@@ -44,16 +36,13 @@
...
 import java.util.List;
 import java.util.Optional;
 
 import static org.apache.commons.lang3.StringUtils.isNotBlank;
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertNotNull;
-import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.*;
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/test/java/eu/europeana/metis/sandbox/service/problempatterns/PatternAnalysisServiceImplIT.java:45]: Fixed

File: src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java (hunk 1 above)
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java:101]: Done

File: src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerIT.java (hunk 2 above)
Comment (JoanaCMS) [reply src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerIT.java:207]: Done

File: src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerTest.java (hunk 4 above)
Comment (JoanaCMS) [reply src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerTest.java:505]: Done

**Pull request merged**